client/extensions/woocommerce/app/order/order-fulfillment/index.js

```diff
@@ -144,34 +144,32 @@ class OrderFulfillment extends Component {
 			translate,
 			hasLabelsPaymentMethod,
 		} = this.props;
-		const isShippable = ! isOrderFinished( order.status );
-		const hideLabels = ! labelsEnabled || ! hasLabelsPaymentMethod;
-		const buttonClassName = classNames( {
-			'is-placeholder': wcsEnabled && ! labelsLoaded,
-		} );
+		const orderFinished = isOrderFinished( order.status );
+		const orderCancelled = 'cancelled' === order.status;
+		const showLabels =
+			wcsEnabled && labelsLoaded && labelsEnabled && hasLabelsPaymentMethod && ! orderCancelled;
+		const labelsLoading = wcsEnabled && ! labelsLoaded && ! orderCancelled;
 
-		if (
-			! isShippable &&
-			( ! wcsEnabled || ( labelsLoaded && hideLabels ) || 'cancelled' === order.status )
-		) {
+		if ( orderFinished && ! labelsLoading && ! showLabels ) {
 			return null;
```

Comment:
`if ( orderCancelled )` must also return `null`. Since `orderFinished` is `true` when the order is cancelled, this case _may_ be covered in this conditional, but that requires some mental gymnastics.